support/Environments/caltech_hpc_gcc_skylake.sh

```diff
@@ -13,12 +13,12 @@ spectre_setup_modules() {
 
 spectre_load_modules() {
     module use /central/groups/sxs/modules/
-    module load libraries/spectre-deps/skylake-2024-02
+    module load libraries/spectre-deps/skylake-2024-03
```

Comment:
Ok removed. I originally had that because the intel MPI module kept printing stuff whenever it was loaded/unloaded which I found annoying, but I found a way to silence it in the intel MPI module file